index.js

```diff
@@ -2,36 +2,68 @@
 
 import { Platform } from 'react-native';
 import StackTrace from 'stacktrace-js';
+import {initSourceMaps, getStackTrace} from 'react-native-source-maps';
 import { Crashlytics } from 'react-native-fabric';
 var assign = require('lodash.assign');
 
-function init() {
+function init(sourcemapName = null) {
   if (__DEV__) {
     // Don't send exceptions from __DEV__, it's way too noisy!
     // Live reloading and hot reloading in particular lead to tons of noise...
     return;
   }
 
   var originalHandler = global.ErrorUtils.getGlobalHandler();
-  function errorHandler(e, isFatal) {
-    StackTrace.fromError(e, {offline: true}).then((x) => {
-      Crashlytics.recordCustomExceptionName(e.message, e.message, x.map((row) => (assign({}, row, {
-        fileName: `${row.fileName}:${row.lineNumber || 0}:${row.columnNumber || 0}`,
-      }))))
-    });
-    // And then re-throw the exception with the original handler
-    if (originalHandler) {
-      if (Platform.OS === 'ios') {
-        originalHandler(e, isFatal);
-      } else {
-        // On Android, throwing the original exception immediately results in the
-        // recordCustomExceptionName() not finishing before the app crashes and therefore not logged
-        // Add a delay to give it time to log the custom JS exception before crashing the app.
-        // The user facing effect of this delay is that separate JS errors will appear as separate
-        // issues in the Crashlytics dashboard.
+  if (sourcemapName) {
+    initSourceMaps({sourceMapBundle: sourcemapName});
+  }
+  async function errorHandler(e, isFatal) {
+    if (sourcemapName) {
+      // map the JS stacktrace using the sourcemap file given on init
+      let jsParsed = false
+      try {
+        e.stack = await getStackTrace(e);
+        if (e.stack) {
+          Crashlytics.recordCustomExceptionName(e.message, e.message, e.stack)
+          jsParsed = true
+        }
+      } catch (error) {
+        // ignore and use the normal stack on the error
+      }
+      // And then re-throw the exception with the original handler
+      if (originalHandler) {
```

Comment:
Should we think about unifying these two? Yours is similar-but-different to what I had done:
- We both have delays, though mine was Android-only. (Did you encounter problems on iOS? Or was this motivated by Android problems?)
- Yours seems to handle fatal errors differently, to avoid duplicates, which might be something that could benefit the original one (which screws that aspect up).

Reply:
I did face problems on the iOS but, it might have been Fabric's delay or not. Given that, a small delay there should not cause trouble at all and worked better for me.
When a JS crash happens, it will log a non-fatal AND a fatal error, because it was the only way to get the JS trace and also make the App actually crash on fatals. 
For non-fatals yes, it logs only once.

Reply:
Makes sense. Do you think you can safely combine your error fallback with the one that existed previously, to avoid duplication?

Reply:
I will try and let you know. I will even try to merge the approaches better so that if there's no stacktrace received from JS, it will use your code.